### PR TITLE
Additional Wake Up / Reset Reason

### PR DESCRIPTION
**Pull request by brolan-juul**
Obtaining more information on reasons why it is resetting. Adding wake up from system off.

### CODE REVIEW

File: hw/hal/include/hal/hal_system.h [hunk 1]
```diff
@@ -66,8 +66,10 @@ enum hal_reset_reason {
     HAL_RESET_SOFT = 4,
     /** Low supply voltage */
     HAL_RESET_BROWNOUT = 5,
+    /** Power on */
+    HAL_RESET_POWER_OFF = 6,
     /** Restart due to user request */
-    HAL_RESET_REQUESTED = 6,
+    HAL_RESET_REQUESTED = 7,
```
Comment (sjanc) [inline hw/hal/include/hal/hal_system.h]: any reason to change HAL_RESET_REQUESTED value? why not just add POWER_OFF at the end of enum?

File: hw/hal/include/hal/hal_system.h [hunk 2]
```diff
@@ -66,8 +66,10 @@ enum hal_reset_reason {
     HAL_RESET_SOFT = 4,
     /** Low supply voltage */
     HAL_RESET_BROWNOUT = 5,
+    /** Power on */
```
Comment (kasjer) [inline hw/hal/include/hal/hal_system.h]: I would change it to HAL_RESET_SYSTEM_OFF or such. Now we have comment stating Power on (which is double confusing:  first see comment on HAL_RESET_POR, second name is HAL_RESET_POWER_OFF)

File: hw/mcu/nordic/nrf52xxx/src/hal_reset_cause.c [hunk 3]
```diff
@@ -37,6 +37,8 @@ hal_reset_cause(void)
         reason = HAL_RESET_SOFT;
     } else if (reg & POWER_RESETREAS_RESETPIN_Msk) {
         reason = HAL_RESET_PIN;
+    } else if(reg & POWER_RESETREAS_OFF_Msk) {
```
Comment (kasjer) [inline hw/mcu/nordic/nrf52xxx/src/hal_reset_cause.c]: space is missing afater if

File: hw/hal/include/hal/hal_system.h (hunk 1 above)
Comment (brolan-juul) [reply hw/hal/include/hal/hal_system.h]: Good point. Moving it back to original lineup.

File: hw/hal/include/hal/hal_system.h (hunk 2 above)
Comment (brolan-juul) [reply hw/hal/include/hal/hal_system.h]: Updated comment to better reflect as to what is happening and to reduce confusion. 

`    /** System Off, wakeup on external interrupt*/`

File: hw/mcu/nordic/nrf52xxx/src/hal_reset_cause.c (hunk 3 above)
Comment (brolan-juul) [reply hw/mcu/nordic/nrf52xxx/src/hal_reset_cause.c]: Good catch. Fixed.

**Comment by brolan-juul** [comment]
@kasjer Noted. Will get to it soon.

**Pull request merged**